### PR TITLE
Set up initial builds

### PR DESCRIPTION
**Pull request by jrryjcksn**
* Set up circle configuration
* The code generation was placing its output in `<path-to-repo>/<path-to-repo>/....` instead of `./<path-to-repo>/....`
* There was no `clean` target.

### CODE REVIEW

**Review by D3nn — APPROVED** [review]
LGTM

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+          command: |
+            echo "The following files have been modified since checkout or are unknown to Git:"
+            ! (git status --porcelain | grep -E '^( M)|(??)')
+      - run:
+          # Allow builds from forks
+          name: Push image if docker login is available
+          command: |
+            if [ -n "$DOCKER_IO_PASSWORD" ]; then
+              docker login -u "$DOCKER_IO_USER" -p "$DOCKER_IO_PASSWORD" docker.io
+              make push
```
Comment (bboreham) [inline .circleci/config.yml:51]: There is no `push` target in the Makefile.

(there is a `docker-push`, but it doesn't seem to do what we want)

Comment (jrryjcksn) [reply .circleci/config.yml:51]: K, thanks.

**Comment by D3nn** [comment]
>
> There is no push target in the Makefile.
> (there is a docker-push, but it doesn't seem to do what we want)


I'll fix that.

**Pull request merged**